### PR TITLE
redefine variables

### LINKED ISSUE #17: Jacobian adjustment due to change of variables.
May need to make a jacobian adjustment to deal with change of variables. Still getting my head around this. 

Some resources (in difficulty order):

- https://mc-stan.org/docs/2_18/reference-manual/change-of-variables-section.html#change-of-variables.section
- https://jsocolar.github.io/jacobians/
- https://rstudio-pubs-static.s3.amazonaws.com/486816_440106f76c944734a7d4c84761e37388.html
- https://betanalpha.github.io/assets/case_studies/probability_theory.html#1_setting_a_foundation

### PR DESCRIPTION
**Pull request by sbfnk**
Closes #17 by redefining variables on transformed scales so as not to have to do a Jacobian correction.

### CODE REVIEW

File: stan/inc2prev.stan [hunk 1]
```diff
@@ -155,11 +161,11 @@ model {
   // Priors for antibody model
   if (n_ab > 0) {
     init_dab ~ normal(init_ab_mean, init_ab_sd);
-    logit(beta) ~ normal(pbeta[1], pbeta[2]);
-    logit(gamma) ~ normal(pgamma_mean, pgamma_sd); 
-    logit(delta) ~ normal(pdelta[1], pdelta[2]);
-    log(k) ~ normal(0, 0.1);
-    log(l) ~ normal(0, 0.1);
+    logit_beta ~ normal(pbeta[1], pbeta[2]);
```
Comment (seabbs) [inline stan/inc2prev.stan:164]: I do think this avoids the issue for the logit space paramters but I am not very clear on why. Why is the LHS of them model block handled differently from the RHS of the TP block. Have you found anything on this that was helpful for getting this?

**Review by seabbs — APPROVED** [review]
I think this resolves the issue if it was present but I am not totally sure I understand why. It might make more sense to use the `lognormal` directly for parameters that support this.

File: stan/inc2prev.stan [hunk 2]
```diff
@@ -91,6 +91,12 @@ transformed parameters {
   array[n_ab] vector[ab_obs] odab;
   array[n] vector[obs] combined_sigma;
   array[n_ab] vector[ab_obs] combined_ab_sigma;
+  vector[n_ab > 0 ? 1 : 0] beta = inv_logit(logit_beta); // tranformation to natural scale
+  vector[n_ab > 0 ? 2 : 0] gamma = inv_logit(logit_gamma); // tranformation to natural scale
+  vector[n_ab > 0 ? 1 : 0] delta = inv_logit(logit_delta); // tranformation to natural scale
+  vector[n_ab > 0 ? 1 : 0] k = exp(log_k); // tranformation to natural scale
```
Comment (seabbs) [inline stan/inc2prev.stan:97]:
```suggestion
  
```

File: stan/inc2prev.stan [hunk 3]
```diff
@@ -91,6 +91,12 @@ transformed parameters {
...
+  vector[n_ab > 0 ? 1 : 0] l = exp(log_l); // tranformation to natural scale
```
Comment (seabbs) [inline stan/inc2prev.stan:98]:
```suggestion

```

File: stan/inc2prev.stan [hunk 4]
```diff
@@ -155,11 +161,11 @@ model {
...
     init_dab ~ normal(init_ab_mean, init_ab_sd);
-    logit(beta) ~ normal(pbeta[1], pbeta[2]);
-    logit(gamma) ~ normal(pgamma_mean, pgamma_sd); 
-    logit(delta) ~ normal(pdelta[1], pdelta[2]);
-    log(k) ~ normal(0, 0.1);
-    log(l) ~ normal(0, 0.1);
+    logit_beta ~ normal(pbeta[1], pbeta[2]);
+    logit_gamma ~ normal(pgamma_mean, pgamma_sd); 
+    logit_delta ~ normal(pdelta[1], pdelta[2]);
+    log_k ~ normal(0, 0.1);
```
Comment (seabbs) [inline stan/inc2prev.stan:167]:
```suggestion
    k ~ lognormal(0, 0.1);
```

File: stan/inc2prev.stan [hunk 5]
```diff
@@ -155,11 +161,11 @@ model {
...
+    log_l ~ normal(0, 0.1);
```
Comment (seabbs) [inline stan/inc2prev.stan:168]:
```suggestion
    l ~ lognormal(0, 0.1);
```

File: stan/inc2prev.stan [hunk 6]
```diff
@@ -73,12 +73,12 @@ parameters {
...
-  vector<lower = 0, upper = 1>[n_ab > 0 ? 1 : 0] beta; // proportion that seroconvert
-  vector<lower = 0, upper = 1>[n_ab > 0 ? 2 : 0] gamma; // antibody waning (inf & vac)
-  vector<lower = 0, upper = 1>[n_ab > 0 ? 1 : 0] delta; // vaccine efficacy
+  vector[n_ab > 0 ? 1 : 0] logit_beta; // proportion that seroconvert
+  vector[n_ab > 0 ? 2 : 0] logit_gamma; // antibody waning (inf & vac)
+  vector[n_ab > 0 ? 1 : 0] logit_delta; // vaccine efficacy
   vector<lower = 0, upper = 1>[n] init_dab; // initial proportion with antibodies
-  vector<lower = 0>[n_ab > 0 ? 1 : 0] k; // Potential loss of efficacy from new infections in already seropositive people
-  vector<lower = 0>[n_ab > 0 ? 1 : 0] l; // Potential loss of efficacy from new doses being administered to already seropositive people
+  vector[n_ab > 0 ? 1 : 0] log_k; // Potential loss of efficacy from new infections in already seropositive people
```
Comment (seabbs) [inline stan/inc2prev.stan:80]:
```suggestion
  vector[n_ab > 0 ? 1 : 0] k; // Potential loss of efficacy from new infections in already seropositive people
```

File: stan/inc2prev.stan [hunk 7]
```diff
@@ -73,12 +73,12 @@ parameters {
...
+  vector[n_ab > 0 ? 1 : 0] log_l; // Potential loss of efficacy from new doses being administered to already seropositive people
```
Comment (seabbs) [inline stan/inc2prev.stan:81]:
```suggestion
  vector[n_ab > 0 ? 1 : 0] l; // Potential loss of efficacy from new doses being administered to already seropositive people
```

File: stan/inc2prev.stan (hunk 1 above)
Comment (sbfnk) [reply stan/inc2prev.stan:164]: I've found these forum posts helpful for understanding the issue and solutions a bit better:
https://discourse.mc-stan.org/t/do-i-need-jacobian-adjustment-here/19875/1
https://discourse.mc-stan.org/t/jacobian-in-reparameterizations/21592
https://discourse.mc-stan.org/t/basic-newbie-question-about-transforming-a-parameter-and-jacobians/1314
https://discourse.mc-stan.org/t/how-do-i-specify-a-logit-normal-distribution/4894/4

File: stan/inc2prev.stan (hunk 4 above)
[the remark this answers is not in the document]
Comment (sbfnk) [reply stan/inc2prev.stan:167]: That seems right, yes.

**Comment by seabbs** [comment]
Ah this was a good resource for this: https://discourse.mc-stan.org/t/basic-newbie-question-about-transforming-a-parameter-and-jacobians/1314/5

**Pull request merged**